frontend/src/pages/RoadDetails.tsx

```diff
@@ -61,27 +72,69 @@ const RoadDetails = () => {
     });
   }, []);
 
+  const handleTopPanelSizeChange = (size: number) => {
+    if (size <= 10) {
+      setTopPanelSize(0);
+      setConditionsGraphSize(100 - (10 / window.innerHeight) * 100);
+    } else if (size >= 90) {
+      setConditionsGraphSize(0);
+      setTopPanelSize(100 - (10 / window.innerHeight) * 100);
+    } else {
+      setTopPanelSize(size);
+      setConditionsGraphSize(100 - size - (10 / window.innerHeight) * 100);
+    }
+  };
+
+  const handleMapAreaSizeChange = (size: number) => {
+    if (size <= 7) {
+      setMapAreaSize(0);
+      setRoadImageSize(100 - (10 / window.innerWidth) * 100);
+    } else if (size >= 93) {
+      setRoadImageSize(0);
+      setMapAreaSize(100 - (10 / window.innerWidth) * 100);
+    } else {
+      setMapAreaSize(size);
+      setRoadImageSize(100 - size - (10 / window.innerWidth) * 100);
+    }
+  };
+
   return (
-    <div>
-      <TopBar isToggleOn={setShowRoadImageMode} />
+    <div className="div-mapAndRoad-container">
```

Comment:
> is the class div-mapAndRoad-container exists in the css files?

> I added it only so when I did the inspect on the page I could recognise the div. should I remove the className?

I would say yes. Also, if it helps, using the React dev tools (the extension) i believe there is a way to go to the component (in the html i mean) :

![image](https://github.com/DTU-SE-Group-D/LiRA-Viz/assets/65665540/b0e84e22-41ae-42d6-b53f-715c22c02201)